.github/workflows/tests_eessi_module.yml

```diff
@@ -58,29 +58,141 @@ jobs:
+          unset EESSI_USE_ARCHSPEC
+          unset EESSI_USE_ARCHDETECT
+          env | grep -E '(^EESSI_|^LMOD_RC|^LMOD_PACKAGE_PATH)' | sort > "${sourceoutfile}"
+          
+          # Now compare the two results
+          echo ""
+          echo "Lmod initialisation:"
           cat "${moduleoutfile}"
+          echo ""
+          echo "Source script initialisation:"
           cat "${sourceoutfile}"
+          echo ""
+          echo ""
           if (diff "${moduleoutfile}" "${sourceoutfile}" > /dev/null); then
               echo "Test for checking env variables PASSED"
           else 
               echo "Test for checking env variables FAILED"  >&2
-              diff "${moduleoutfile}" "${sourceoutfile}" 
+              diff --unified=0 "${moduleoutfile}" "${sourceoutfile}" 
               exit 1
           fi
 
+  make_sure_load_and_unload_work:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        EESSI_VERSION:
+          - 2023.06
+        EESSI_SOFTWARE_SUBDIR_OVERRIDE:
+          - none
+          - x86_64/amd/zen2
+          - x86_64/amd/zen4
+    steps:
+      - name: Check out software-layer repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+          
+      - name: Mount EESSI CernVM-FS pilot repository
+        uses: cvmfs-contrib/github-action-cvmfs@55899ca74cf78ab874bdf47f5a804e47c198743c # v4.0
+        with:
+          cvmfs_config_package: https://github.com/EESSI/filesystem-layer/releases/download/latest/cvmfs-config-eessi_latest_all.deb
+          cvmfs_http_proxy: DIRECT
+          cvmfs_repositories: software.eessi.io
+              
+      - name: Test for identical environment after loading and unloading the EESSI module
+        run: |
+          . /cvmfs/software.eessi.io/versions/${{matrix.EESSI_VERSION}}/compat/linux/$(uname -m)/usr/share/Lmod/init/bash  # Initialise Lmod
+          
+          # Set our path overrides according to our matrix
+          if [[ "${{matrix.EESSI_SOFTWARE_SUBDIR_OVERRIDE}}" != "none" ]]; then
+              export EESSI_SOFTWARE_SUBDIR_OVERRIDE=${{matrix.EESSI_SOFTWARE_SUBDIR_OVERRIDE}}
+          fi
+
+          # Turn on debug output in case we want to take a look
+          export EESSI_DEBUG_INIT=true
+
+          initial_env_file="initial_env.txt"
+          module_cycled_file="load_unload_cycle.txt"
+          
+          # Store the initial environment
+          env | grep -E '(^EESSI_|^LMOD_RC|^LMOD_PACKAGE_PATH)' | sort > "${initial_env_file}"
+
+          # Do (and undo) loading the EESSI module
+          export MODULEPATH=init/modules
+          CPU_ARCH=$(./init/eessi_archdetect.sh -a cpupath)
+          module load EESSI/${{matrix.EESSI_VERSION}}
+          module unload EESSI/${{matrix.EESSI_VERSION}}
+          env | grep -E '(^EESSI_|^LMOD_RC|^LMOD_PACKAGE_PATH)' | sort > "${module_cycled_file}"
```

Comment:
While it's fine to check only this subset of env-vars, I'm wondering if there is any downside to checking _all_. The only thing that would require is do also undo your `MODULEPATH` change from a few lines above, but that's easy enough (store in an `ORIGINAL_MODULEPATH` and restore later).

Reply:
The same could btw be argued for the `init` vs `module` test, though I think there might be more changes to the environment there that would make such a comparison of the entire env harder.

Reply:
In this case, yes, we could (and should) check the entire environment. For the script to Lmod comparision, Lmod variables related to loaded modules will be different, so rather than an include list you'd need an exclude list...which to me is more painful to maintain (since we know what we set, but we have to figure out what Lmod sets).

Reply:
Done in https://github.com/EESSI/software-layer/pull/781/commits/6e1ccaa36e028b81e9a58b1958ca2f3102ba1a1b